runtime/src/iree/base/allocator.c

```diff
@@ -88,6 +88,7 @@ static iree_status_t iree_allocator_system_alloc(
     } else if (command == IREE_ALLOCATOR_COMMAND_CALLOC) {
       IREE_TRACE_ZONE_BEGIN_NAMED(z0_named, "iree_allocator_system_calloc");
       z0 = z0_named;
+      IREE_TRACE_ZONE_APPEND_VALUE_I64(z0, byte_length);
```

Comment:
Do you want to keep this value, or was this just added for testing? This could add a (little) bit of overhead.

For memory allocation tracking, we do also have `IREE_TRACING_FEATURE_ALLOCATION_TRACKING` that integrates with Tracy's memory support.

Reply:
I think it is a generally important but if information for calloc: I was trying to see why some were taking a long time and it turned out they were zeroing hundreds of KB.

Reply:
Tracy gives allocation information without adding this user value:
![image](https://github.com/openxla/iree/assets/75337/b1938cc2-43cb-430e-b364-21f94a1c0502)
(note "memory allocated")

On allocations we expect to be big (like heap HAL buffers) we already tag the size so you can do things like group by allocation size and sort by MTPC:
![image](https://github.com/openxla/iree/assets/75337/7ce5e8f5-757f-40a2-9621-9eafa5e021d8)
For the big HAL heap allocations it's best to look at the HAL functions so you don't see all the noise from other things. 

Only reason I mention this is because to keep trace files small/the overhead minimal we try not to include redundant information. The only benefit here is that you can see the size in the tool tip in the timeline, though you can also see it in the memory plot on the timeline. Ok with having it if the tooltip makes/breaks your workflow, but would really encourage not duplicating things :)

(I've thought about adding a tracy UI mode where holding shift/alt/etc while hovering on the timeline would show more details - like allocations - without having to pop up the zone info)

Reply:
I'm fine popping it out. It was just the result of having spent a number of minutes trying to figure out why some of these were more expensive than others, and then I left it in there. Knowing how to hold it is fine.